stdlib/array.ml

```diff
@@ -442,8 +442,9 @@ let fast_sort = stable_sort
 let shuffle ~rand a = (* Fisher-Yates *)
   for i = length a - 1 downto 1 do
     let j = rand (i + 1) in
+    if not (0 <= j && j <= i) then invalid_arg "Array.shuffle";
```

Comment:
How about this version instead?
`if j < 0 || j > i then invalid_arg "Array.shuffle";`